docs/src/advanced_topics/execution_trace_optimization.md

```diff
@@ -0,0 +1,44 @@
+# Execution trace optimization
+
+## Understanding cycle counts in Miden VM
+
+When we refer to "number of cycles" in most Miden VM documentation, we're specifically referring to the **stack rows** portion of the execution trace. However, the actual proving time is determined by what we call the "true number of cycles," which is the maximum of all trace segment lengths:
+
+- **Stack rows**: One row per VM operation (what `clk` outputs)
+- **Range checker rows**: Added when opcodes request 16-bit range checks
+- **Chiplet rows**: Added when opcodes call specialized chiplets
```

Comment:
Would give some intuition here, so:
- `hperm` calls the hasher chiplet
- `and`, `or` (and other bitwise ops) call the bitwise chiplet
- memory operations call the memory chiplet
- syscalls call the kernel ROM chiplet